### PR TITLE
Add support for post__in and post__not_in

### PR DESCRIPTION
**Pull request by sc0ttkclark**
Wasn't sure what to put for `@since` but I can tweak it before merge or you can do it after.

### CODE REVIEW

**Comment by sc0ttkclark** [comment]
Once I get this one looked at and merged (or rejected lol), I'll submit the additional patch for post__not_in:

```php
		/**
		 * 'post__not_in' arg support.
		 *
		 * @since x.x
		 */
		if ( ! empty( $args[ 'post__not_in' ] ) ) {
			$filter['and'][]['bool']['must_not'] = array(
				'terms' => array(
					'post_id' => (array) $args[ 'post__not_in' ],
				),
			);

			$use_filters = true;
		}
```

**Comment by tlovett1** [comment]
Can you add a unit test for this one, Scott? Also, is there a huge advantage in grabbing the posts from ES instead of MySQL?

**Comment by sc0ttkclark** [comment]
post__in but with a meta_query / tax_query, that's really the only advantage there.

I'll get to adding unit tests for this, and start prepping all of the other PRs (with unit tests too).

**Comment by tlovett1** [comment]
Makes sense.

**Comment by tlovett1** [comment]
@sc0ttkclark can you finish this please?

**Comment by sc0ttkclark** [comment]
Tests added, will watch for pass/fail

**Comment by sc0ttkclark** [comment]
Runs, and new tests pass :)

**Comment by sc0ttkclark** [comment]
Note: This code supports post__in and post__not_in, I merged them into the PR as one.

File: classes/class-ep-api.php [hunk 1]
```diff
@@ -755,6 +755,36 @@ public function format_args( $args ) {
...
 
 		/**
+		 * 'post__in' arg support.
+		 *
+		 * @since x.x
+		 */
+		if ( ! empty( $args['post__in'] ) ) {
+			$filter['and'][]['bool']['must'] = array(
+				'terms' => array(
+					'post_id' => (array) $args['post__in'],
```
Comment (tlovett1) [inline classes/class-ep-api.php]: Not a huge fan of the array cast since I believe `(array) false === array( '' )`

**Comment by tlovett1** [comment]
@sc0ttkclark last request... Can you add this to the documentation :)

**Pull request merged**